chromium_src/chrome/browser/extensions/api/cookies/cookies_api.cc

```diff
@@ -0,0 +1,23 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "chrome/browser/extensions/api/cookies/cookies_api.h"
+
+// This disables cookies.onChange routing in Tor windows (as it worked
+// when there was a DCHECK instead of CHECK). Once crbug.com/417228685 is fixed
+// in upstream, we'll be able to manage Tor profiles like any other main-OTR
+// profile.
+
+#define IsSerializeable(...) IsSerializeable(__VA_ARGS__) && !profile_->IsTor()
+
+#define BRAVE_EXTENSIONS_COOKIES_EVENT_ROUTER_ON_OFF_THE_RECORD_PROFILE_CREATED \
+  if (off_the_record->IsTor()) {                                                \
+    return;                                                                     \
+  }
```

Comment:
can we use the same define method same as `IsSerializeable` so we don't have add additional patch?
```
#define IsPrimaryOTRProfile(... ) IsPrimaryOTRProfile(__VA_ARGS__) || off_the_record->IsTor()
```